lib/tenstorrent/bh_arc/gddr.c

```diff
@@ -290,9 +294,61 @@ static void wipe_l1(void)
 				uint8_t x, y;
 
 				GetGddrNocCoords(gddr_inst, noc2axi_port, noc_id, &x, &y);
-				/* AXI enable must not be set, using MRISC address 0 */
-				noc_dma_write(tensix_x, tensix_y, addr, x, y, addr, MRISC_L1_SIZE,
-					      true);
+
+				/* Use Zephyr DMA driver to write from tensix to GDDR MRISC L1 */
+				struct tt_bh_dma_noc_coords coords = {
+					.source_x = tensix_x,
+					.source_y = tensix_y,
+					.dest_x = x,
+					.dest_y = y,
+				};
```

Comment:
Something like this is ripe for a `static inline` function (or a macro) that constructs it.

E.g.
```cpp
#define TT_BH_DMA_NOC_COORD_INIT(sx, sy, dx, dy) \
(struct tt_bh_dma_noc_coords) { \
  .source_x = (sx), \
  .source_y = (sy), \
  .dest_x = (dx), \
  .dest_y = (dy), \
}

#define TT_BH_DMA_NOC_COORD(name, sx, sy, dx, dy) \
struct tt_bh_dma_noc_coords name = TT_BH_DMA_NOC_COORD_INIT(sx, sy, dx, dy)

/* IIRC, uint8_t is ok? */
static inline struct tt_bh_dma_noc_coords tt_bh_dma_noc_coord_init(uint8_t sx, uint8_t sy, uint8_t dx, uint8_t dy) {
  return TT_BH_DMA_NOC_COORD_INIT(sx, sy, dx, dy);
}
```